test/stdlib/POSIX.swift

```diff
@@ -17,21 +13,59 @@ chdir(CommandLine.arguments[1])
 var POSIXTests = TestSuite("POSIXTests")
 
 let semaphoreName = "TestSem"
+#if os(Android)
+// In Android, the cwd is the root directory, which is not writable.
+let fn: String = {
+  let capacity = Int(PATH_MAX)
+  let resolvedPath = UnsafeMutablePointer<Int8>.allocate(capacity: capacity)
+  resolvedPath.initialize(repeating: 0, count: capacity)
+  defer {
+    resolvedPath.deinitialize(count: capacity)
+    resolvedPath.deallocate()
+  }
+  guard let _ = realpath("/proc/self/exe", resolvedPath) else {
+    fatalError("Couldn't obtain executable path")
+  }
+
+  let length = strlen(resolvedPath)
+  guard length != 0 else {
```

Comment:
I believe this "guard and fatalError" pattern can be expressed by a single call.
```swift
precondition(length != 0, "Couldn't obtain a valid executable path")
```